File: perl/dave_game.pl
use warnings;

my $range = 10;
my $random_number = int(rand($range));

while("true")
{
    print "Guess a number (0 - 10):\n";

    $guess = <STDIN>;
    chop $guess;

    if ($guess > $random_number) {
        print "Too high! Try again.\n";
    }
    elsif ($guess < $random_number) {
        print "Too low! Try again.\n";
    }
    elsif ($guess == $random_number) {
        print "You guessed it! Goodbye.\n";
        exit;
    }
    else {
        print "That's not a valid number."
    }
}

File: ruby/dave_game.rb
random_number = rand(1 + 10)

while true
  puts "Guess a number (0 - 10):"

  guess = gets.chomp.to_i
  if guess > random_number
    puts "Too high! Try again."
  elsif guess < random_number
    puts "Too low! Try again."
  elsif guess == random_number
    puts "You guessed it! Goodbye."
    exit
  else
    puts "That's not a valid number."
  end
end